woocommerce/woocommerce-gutenberg-products-block
# E2E: Refactor Mini Cart to be ready for fully parallel

## What

This PR:
- refactors the Mini Cart test to remove the usage of CSS selector in favor of role based locators.
- makes the tests fully isolated.
- adds a new `useIncognito()` util for running tests in isolation.

## Why



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Manually update the `tests/e2e/playwright.config.ts`:
```js
	retries: CI ? 2 : 0,
	// workers: 1, // Comment out this line
	fullyParallel: true, // Add this line
	reporter: process.env.CI
```
2. Run this command `npm run test:e2e tests/e2e/tests/mini-cart/mini-cart.block_theme.spec.ts`.
3. See the tests run in parallel and passing:

```sh
% npm run test:e2e tests/e2e/tests/mini-cart/mini-cart.block_theme.spec.ts

> @woocommerce/block-library@11.0.0-dev test:e2e
> sh ./bin/check-env.sh && npx playwright test --config=tests/e2e/playwright.config.ts

Customer state file deleted successfully.
Logged-in as customer successfully.

Running 4 tests using 4 workers

  ✓  1 [blockTheme] › mini-cart/mini-cart.block_theme.spec.ts:57:6 › Mini Cart Block › should open the filled cart drawer (5.8s)
  ✓  2 [blockTheme] › mini-cart/mini-cart.block_theme.spec.ts:23:6 › Mini Cart Block › should close the drawer when clicking on the close button (5.4s)
  ✓  3 [blockTheme] › mini-cart/mini-cart.block_theme.spec.ts:38:6 › Mini Cart Block › should close the drawer when clicking outside the drawer (7.2s)
  ✓  4 [blockTheme] › mini-cart/mini-cart.block_theme.spec.ts:14:6 › Mini Cart Block › should open the empty cart drawer (3.8s)

  4 passed (7.6s)
```

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Add suggested changelog entry here.
